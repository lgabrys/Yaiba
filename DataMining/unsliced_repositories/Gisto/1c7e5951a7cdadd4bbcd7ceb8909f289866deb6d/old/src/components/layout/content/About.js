import React from 'react';
import styled from 'styled-components';

import { get } from 'lodash/fp';

import { baseAppColor, bg } from 'constants/colors';

import { isElectron } from 'utils/electron';

import ExternalLink from 'components/common/ExternalLink';
import Anchor from 'components/common/Anchor';

import * as packageJson from '../../../../package.json';
import logoImg from '../../../../build/icon.png';

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  color: #555;
  height: 100%;
  width: 50%;
  transform: translate(50%);
  justify-content: center;
  text-align: left;
`;

const UpdateInfo = styled.div`
  border: 1px solid ${baseAppColor};
  padding: 10px;
  background: ${bg};
  border-radius: 3px;
`;

export class About extends React.Component {
  state = {
    message: ''
  };

  componentDidMount() {
    if (isElectron) {
      const { ipcRenderer } = require('electron');

      ipcRenderer.send('checkForUpdate');
      ipcRenderer.on('updateInfo', (event, text, info) => {
        const url = get('url[0].browser_download_url', info);

        if (url) {
          const message = (
            <React.Fragment>
              <strong>{text}</strong>
              &nbsp;
              <Anchor href={ url }>Download</Anchor>
            </React.Fragment>
          );

          this.setState({ message });
        } else {
          this.setState({ message: text });
        }
      });
    }
  }

  render() {
    return (
      <Wrapper>
        <img src={ logoImg } width="80" alt="" />
        <h2>About Gisto</h2>
        <p>
          Current version <strong>v{packageJson.version}</strong>
        </p>

        {this.state.message && <UpdateInfo>{this.state.message}</UpdateInfo>}

        <p>
          Gisto is a code snippet manager that runs on GitHub Gists and adds additional features
          such as searching, tagging and sharing gists while including a rich code editor.
        </p>

        <p>
          All your data is stored on GitHub and you can access it from GitHub Gists at any time with
          changes carrying over to Gisto.
        </p>

        <p>
          <ExternalLink href="https://github.com/Gisto/Gisto/blob/master/CHANGELOG.md">Change log</ExternalLink>
          &nbsp;|&nbsp;
          <ExternalLink href="https://github.com/Gisto/Gisto">GitHub</ExternalLink>
          &nbsp;|&nbsp;
          <ExternalLink href="https://gistoapp.com">Website gistoapp.com</ExternalLink>
          &nbsp;|&nbsp;
          <ExternalLink href="https://github.com/Gisto/Gisto/issues">Issues</ExternalLink>
          &nbsp;|&nbsp;
          <ExternalLink href="https://twitter.com/gistoapp">Twitter</ExternalLink>
        </p>
      </Wrapper>
    );
  }
}

About.propTypes = {};

export default About;
