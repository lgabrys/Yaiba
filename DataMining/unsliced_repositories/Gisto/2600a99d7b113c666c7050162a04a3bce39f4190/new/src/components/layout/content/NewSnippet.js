import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import styled from 'styled-components';
import {
  filter,
  findIndex,
  get,
  head,
  isEmpty,
  keys,
  map,
  set,
  replace,
  startsWith,
  debounce
} from 'lodash/fp';
import uuid from 'uuid';
import { HashRouter as Router, Link } from 'react-router-dom';

import * as snippetActions from 'actions/snippets';

import { DEFAULT_SNIPPET_DESCRIPTION } from 'constants/config';
import { syntaxMap } from 'constants/editor';

import { prepareFiles } from 'utils/snippets';
import { gaPage } from 'utils/ga';
import { getSetting } from 'utils/settings';
import { randomString } from 'utils/string';
import { hexToRGBA } from 'utils/color';

import Input from 'components/common/controls/Input';
import Editor from 'components/common/controls/Editor';
import Icon from 'components/common/Icon';
import ExternalLink from 'components/common/ExternalLink';
import Button from 'components/common/controls/Button';
import Checkbox from 'components/common/controls/Checkbox';
import DropZone from 'components/common/DropZone';
import Select from 'react-dropdown-select';
import { getTags } from 'selectors/snippets';

export class NewSnippet extends React.Component {
  state = {
    public: getSetting('defaultNewIsPublic', false),
    description: '',
    files: [],
    tags: []
  };

  componentDidMount() {
    gaPage('Add new');
    this.addFile();
  }

  setDescription = debounce(300, (description) => {
    this.setState({ description });
  });

  setFileData = (value, id, type) => {
    const { files } = this.state;
    const indexofFile = findIndex({ uuid: id }, files);
    const newFiles = set([indexofFile, type], value, files);

    this.setState({ files: newFiles });
  };

  setFileDataDebounced = debounce(300, this.setFileData);

  togglePublic = () => {
    this.setState((prevState) => ({
      public: !prevState.public
    }));
  };

  addFile = (name, content = '') => {
    const fileStructure = {
      uuid: uuid.v4(),
      name: name || randomString(5),
      language: getSetting('setings-default-new-snippet-language', 'Text'),
      content
    };

    this.setState((prevState) => ({
      files: [...prevState.files, fileStructure]
    }));
  };

  deleteFile = (id) => {
    const { files } = this.state;
    const updatedFiles = filter((file) => file.uuid !== id, files);

    this.setState({ files: updatedFiles });
  };

  save = () => {
    this.props.createSnippet({
      description: `${this.state.description || DEFAULT_SNIPPET_DESCRIPTION} ${map(
        'value',
        this.state.tags
      ).join(' ')}`,
      isPublic: this.state.public,
      files: prepareFiles(this.state.files)
    });
  };

  seTags = (tagList) => {
    const tags = map((tag) => {
      if (!startsWith('#', tag.value)) {
        return set('value', `#${tag.value}`, tag);
      }

      return tag;
    }, tagList);

    this.setState({ tags });
  };

  mapArrayToSelectObject = (array) => map((key) => ({ label: key, value: key }), array);

  customDropdownRenderer = ({ props, state, methods }) => {
    const regexp = new RegExp(state.search, 'i');

    return (
      <div>
        <SearchAndToggle color={ props.color }>
          <input
            type="text"
            value={ state.search }
            autoFocus
            onChange={ methods.setSearch }
            placeholder="Find language"/>
        </SearchAndToggle>
        <Items>
          {props.options
            .filter((item) => regexp.test(item[props.searchBy] || item[props.labelField]))
            .map((option) => {
              if (!props.keepSelectedInList && methods.isSelected(option)) {
                return null;
              }

              return (
                <Item
                  disabled={ option.disabled }
                  key={ option[props.valueField] }
                  onClick={ option.disabled ? null : () => methods.addItem(option) }>
                  <ItemLabel>{option[props.labelField]}</ItemLabel>
                </Item>
              );
            })}
        </Items>
      </div>
    );
  };

  render() {
    const { theme, tags } = this.props;
    const languageFromSettings = getSetting('setings-default-new-snippet-language', 'Text');

    return (
      <Wrapper>
        <Side>
          <SideInner>
            <DropZone onAddFile={ this.addFile }/>

            <H1>
              <strong>New {this.state.public ? 'public' : 'private'} snippet:</strong>{' '}
              {this.state.description}
            </H1>

            <DescriptionSection>
              <StyledInput
                type="text"
                onChange={ (event) => this.setDescription(event.target.value) }
                placeholder={ `Description (default ${DEFAULT_SNIPPET_DESCRIPTION})` }/>
              <StyledSelect
                multi
                create
                style={ { zIndex: 2 } }
                createNewLabel="add '{search}' tag"
                values={ this.state.tags }
                options={ map(
                  (tag) => ({
                    label: startsWith('#', tag) ? replace('#', '', tag) : tag,
                    value: tag
                  }),
                  tags
                ) }
                color={ theme.baseAppColor }
                keepSelectedInList={ false }
                dropdownHeight="200px"
                addPlaceholder="+ Add more"
                placeholder="Add tags"
                onChange={ (values) => this.seTags(values) }/>
            </DescriptionSection>

            <Section>
              <StyledCheckbox
                checked={ this.state.public }
                value={ getSetting('defaultNewIsPublic', false) }
                onChange={ () => this.togglePublic() }/>
              &nbsp;
              <span>
                Public snippet &nbsp;
                <ExternalLink href="https://help.github.com/articles/about-gists/#types-of-gists">
                  <Icon type="info" size="16" color={ theme.baseAppColor }/>
                </ExternalLink>
              </span>
            </Section>

            <ButtonsSection>
              <Router>
                <StyledButton icon="arrow-left" invert>
                  <StyledLink to="/">Back to list</StyledLink>
                </StyledButton>
              </Router>

              <StyledButton invert icon="add" onClick={ () => this.addFile() }>
                Add file
              </StyledButton>

              <StyledButton
                icon="success"
                onClick={ () => this.save() }
                disabled={ isEmpty(this.state.files) }>
                Save
              </StyledButton>
            </ButtonsSection>
          </SideInner>
        </Side>
        <Files>
          {map(
            (file) => (
              <FileSection key={ file.uuid }>
                <div>
                  <StyledFileName
                    type="text"
                    value={ file.name }
                    onChange={ (event) =>
                      this.setFileDataDebounced(event.target.value, file.uuid, 'name')
                    }
                    placeholder="file.ext"/>
                  <StyledSelect
                    values={ [
                      {
                        label: languageFromSettings,
                        value: languageFromSettings
                      }
                    ] }
                    color={ theme.baseAppColor }
                    contentRenderer={ ({ state }) => (
                      <div>{state.values && state.values[0].label}</div>
                    ) }
                    dropdownRenderer={ this.customDropdownRenderer }
                    placeholder="Select language"
                    options={ this.mapArrayToSelectObject(keys(syntaxMap)) }
                    onChange={ (value) =>
                      this.setFileDataDebounced(get('value', head(value)), file.uuid, 'language')
                    }/>
                  <StyledDeleteButton
                    icon="delete"
                    invert
                    onClick={ () => this.deleteFile(file.uuid) }>
                    <strong>Remove</strong> {file.name || 'this file'}
                  </StyledDeleteButton>
                </div>
                <br/>
                <br/>

                <Editor
                  file={ file }
                  isNew
                  id={ file.uuid }
                  onChange={ (value) => this.setFileDataDebounced(value, file.uuid, 'content') }/>
              </FileSection>
            ),
            this.state.files
          )}
        </Files>
      </Wrapper>
    );
  }
}

const mapStateToProps = (state) => ({
  theme: get(['ui', 'settings', 'theme'], state),
  tags: getTags(state)
});

NewSnippet.propTypes = {
  createSnippet: PropTypes.func,
  theme: PropTypes.object,
  tags: PropTypes.array
};

export default connect(
  mapStateToProps,
  {
    createSnippet: snippetActions.createSnippet
  }
)(NewSnippet);

const Wrapper = styled.div`
  display: flex;
`;
const Side = styled.div`
  width: 310px;
  height: calc(100vh - 157px);
  position: relative;
`;
const SideInner = styled.div`
  position: fixed;
  width: 275px;
  height: calc(100vh - 157px);
`;

const Files = styled.div`
  width: calc(100% - 147px);
  margin: -20px 0 0 30px;
`;

const StyledInput = styled(Input)`
  margin: 0;
  text-indent: 10px;
  width: 100%;
  z-index: 1;
`;

const StyledFileName = styled(StyledInput)`
  width: 40%;
  flex: 1;
  margin: 0 20px 0 0;
`;

const StyledCheckbox = styled(Checkbox)`
  margin: 0 10px 0 0;
`;

const Section = styled.div`
  margin: 20px 0;
  flex-direction: column;
`;

const DescriptionSection = styled(Section)`
  display: flex;
`;

const FileSection = styled(Section)`
  border: 1px solid ${(props) => props.theme.baseAppColor};
  padding: 20px;
  border-radius: 3px;

  > div {
    display: flex;
    justify-content: space-between;
    align-items: center;
    height: 0;
    margin: 10px 0;
  }

  &:last-of-type {
    height: calc(100vh - 198px);
  }
`;

const StyledButton = styled(Button)``;

const StyledDeleteButton = styled(StyledButton)`
  line-height: 21px;
  margin: 0 0 0 20px;
  width: auto;
  white-space: nowrap;
  color: ${(props) => props.theme.colorDanger};
  border-color: ${(props) => props.theme.colorDanger};

  span {
    background-color: ${(props) => props.theme.colorDanger};
  }
`;

const H1 = styled.h1`
  font-weight: 300;
  font-size: 22px;
`;

const ButtonsSection = styled.section`
  position: absolute;
  bottom: 0;
  padding: 20px 0 0;
  margin: 0;
  display: flex;
  justify-content: space-between;
  width: 100%;
  z-index: 1;
`;

const StyledLink = styled(Link)`
  color: ${(props) => props.theme.baseAppColor};
  text-decoration: none;
  line-height: 25px;
`;

const StyledSelect = styled(Select)`
  background: #fff;
  border: none !important;
  border-bottom: 1px solid ${(props) => props.theme.baseAppColor} !important;
  border-radius: 0 !important;
  padding: 0 10px;
  min-height: 28px !important;
  z-index: 2;
  margin: 20px 0;
  width: 267px !important;
`;

const SearchAndToggle = styled.div`
  display: flex;
  flex-direction: column;

  input {
    margin: 10px 10px 0;
    line-height: 30px;
    padding: 0px 10px;
    border: 1px solid #ccc;
    border-radius: 3px;
    :focus {
      outline: none;
    }
  }
`;

const Items = styled.div`
  overflow: auto;
  min-height: 10px;
  max-height: 200px;
`;

const Item = styled.div`
  display: flex;
  align-items: baseline;
  ${({ disabled }) => disabled && 'text-decoration: line-through;'};
  cursor: pointer;

  :hover {
    background: ${(props) => hexToRGBA(props.theme.baseAppColor, 0.1)};
  }

  :first-child {
    margin-top: 10px;
  }

  :last-child {
    margin-bottom: 10px;
  }
`;

const ItemLabel = styled.div`
  margin: 5px 10px;
`;
