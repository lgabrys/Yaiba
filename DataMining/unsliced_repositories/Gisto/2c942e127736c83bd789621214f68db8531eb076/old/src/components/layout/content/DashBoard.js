import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import styled, { css } from 'styled-components';
import {
  get, size, filter, map, flow, flattenDeep, uniq, compact, isEmpty 
} from 'lodash/fp';
import { HashRouter as Router, NavLink } from 'react-router-dom';

import {
  getSnippets, getStarredCount, getLanguages, getPrivate 
} from 'selectors/snippets';

import {
  baseAppColor, borderColor, headerBgLightest, lightText 
} from 'constants/colors';
import { DEFAULT_SNIPPET_DESCRIPTION } from 'constants/config';

import * as snippetActions from 'actions/snippets';

import Icon from 'components/common/Icon';
import Input from 'components/common/controls/Input';
import ScrollPad from 'react-scrollpad';
import { gaPage } from 'utils/ga';

const DashbordWrapper = styled.div`
  display: grid;
  grid-template-columns: 3fr 3fr 3fr 3fr;
  grid-gap: 30px;
  color: #444;
  height: 100%;
  h3 {
    margin: 9px 0 20px;
    color: ${baseAppColor};
    font-weight: 300;
    font-size: 16px;
  }
`;

const ContainerWithPills = `
  display: grid;
  grid-gap: 10px;
  grid-template-columns: repeat(auto-fill, minmax(80px, 1fr) ) ;
  color: ${baseAppColor};
  max-height: 50vh;
  overflow: auto;
  font-size: smaller;
  cursor: pointer;
`;

const Box = styled.div`
  background: ${lightText};
  padding: 20px;
  border-radius: 3px;
  box-shadow: 0 0 10px ${borderColor};
`;

const gridBoxInnerCss = css`
  h3 {
    font-size: 22px;
    margin: 0 0 10px 0;
  }
  
  span {
    font-size: 42px;
    float: right;
    color: ${baseAppColor};
    
    small {
      font-size: 10px;
      float: right;
    }
  }
  
  :hover {
    box-shadow: 0 0 50px ${borderColor};
    cursor: pointer;
  }
`;

const Private = styled(Box)`
  grid-column-start: 2;
  grid-column-end: 3;
  grid-row-start: 1;
  grid-row-end: 1;
  
${gridBoxInnerCss};
`;

const Public = styled(Box)`
  grid-column-start: 1;
  grid-column-end: 2;
  grid-row-start: 1;
  grid-row-end: 1;
  
  ${gridBoxInnerCss};
`;

const Starred = styled(Box)`
  grid-column-start: 3;
  grid-column-end: 4;
  grid-row-start: 1;
  grid-row-end: 1;
  
  ${gridBoxInnerCss};
`;

const Untitled = styled(Box)`
  grid-column-start: 4;
  grid-column-end: 5;
  grid-row-start: 1;
  grid-row-end: 1;
  
  ${gridBoxInnerCss};
`;

const Language = styled(Box)`
  grid-column-start: 1;
  grid-column-end: 3;
  grid-row-start: 2;
  grid-row-end: 2;
  max-height: 40vh;
  > div {
    ${ContainerWithPills};
    max-height: 35vh;
    overflow: auto;
  }
  
  strong {
    //float: right;
  }
`;

const Stars = styled(Box)`
  grid-column-start: 3;
  grid-column-end: 5;
  grid-row-start: 2;
  grid-row-end: 2;
  max-height: 40vh;
  > ul {
    overflow: auto;
    max-height: 35vh;
    list-style-type: none;
    padding: 0;
    margin-top: 0;
    
    li {
      margin: 10px 0;
      position:relative;
      overflow: hidden;
      white-space: nowrap;
      &:after {
      content: "";
      width: 100px;
      height: 50px;
      position: absolute;
      top: 0;
      right: 0;
      background: -webkit-gradient(linear, left top, right top, color-stop(0%, rgba(255, 255, 255, 0)), color-stop(56%, rgba(255, 255, 255, 1)), color-stop(100%, rgba(255, 255, 255, 1)));
      background: -webkit-linear-gradient(left, rgba(255, 255, 255, 0) 0%, rgba(255, 255, 255, 1) 56%, rgba(255, 255, 255, 1) 100%);
    }
    }
  }
`;

const Tags = styled(Box)`
  grid-column-start: 1;
  grid-column-end: 5;
  grid-row-start: 3;
  grid-row-end: 3;
  > div {
    ${ContainerWithPills}
  }
`;

const Pill = styled.span`
  border: 1px solid ${headerBgLightest};
  color: ${baseAppColor};
  padding: 5px;
  border-radius: 3px;
  &:hover {
    border: 1px solid ${baseAppColor};
  }
`;

const HeadingWithSearch = styled.span`
    display: flex;
    justify-content: space-between;
    align-items: baseline;
`;

const StyledInput = styled(Input)`
    width: 15vw;
    margin: 0 10px;
`;

const StyledNavLink = styled(NavLink)`
  text-decoration: none;
  color: ${baseAppColor};
  
  :hover {
    color: ${headerBgLightest};
  }
`;

export class DashBoard  extends React.Component {
  state = {
    searchTags: '',
    searchStarred: ''
  };

  componentDidMount() {
    this.props.getRateLimit();
    gaPage('Dashboard');
  }


  getStarred = () => {
    const starred = filter({ star: true }, this.props.snippets);

    if (!isEmpty(this.state.searchStarred)) {
      return filter((starredSnippet) => {
        const regex = new RegExp(this.state.searchStarred, 'gi');

        return starredSnippet.description.match(regex);
      }, starred);
    }

    return starred;
  };

  getUntitled = () => flow([
    filter({ description: DEFAULT_SNIPPET_DESCRIPTION }),
    size
  ])(this.props.snippets);

  getTags = () => {
    const tags = map('tags', this.props.snippets);

    const tagList = flow([
      flattenDeep,
      uniq,
      compact
    ])(tags);

    if (!isEmpty(this.state.searchTags)) {
      return filter((tag) => {
        const regex = new RegExp(this.state.searchTags, 'gi');

        return tag.match(regex);
      }, tagList.sort());
    }

    return tagList.sort();
  };

  linearGradient = (percentOf) => {
    const percents = (percentOf / size(this.props.snippets)) * 100;

    return {
      background: `linear-gradient(to right, ${headerBgLightest} ${Math.floor(percents)}%, #fff ${Math.floor(percents)}%)`
    };
  };

  searchTags = (value) => this.setState({
    searchTags: value
  });

  searchStarred = (value) => this.setState({
    searchStarred: value
  });

  renderLanguages = () => map((languageItem) => {
    const language = get('language', languageItem);
    const filesCount = get('size', languageItem);

    return (
      <Pill style={ this.linearGradient(filesCount) }
              key={ language }
              onClick={ () => this.props.searchByLanguages(language) }>
        {language || 'Other'}
        <br/>
        <strong>{filesCount}</strong> <small>files</small>
      </Pill>
    );
  }, this.props.snippetsLanguages);

  renderTags = () => map((tag) => (
    <Pill key={ tag } onClick={ () => this.props.searchByTags(tag) }>
      { tag }
    </Pill>
  ), this.getTags());

  renderStarred = () => map((snippet) => (
    <li key={ snippet.id }>
      <Icon type={ snippet.public ? 'unlock' : 'lock' } color={ baseAppColor }/>
      &nbsp;
      <Router>
        <StyledNavLink exact
                       className="link"
                       activeClassName="selected"
                       title={ snippet.description }
                       to={ `/snippet/${snippet.id}` }>
          { snippet.description }
        </StyledNavLink>
      </Router>
    </li>
  ), this.getStarred());

  render() {
    const {
      snippets, privateSnippets, starred, searchByStatus
    } = this.props;
    const publicSnippetsCount = size(snippets) - privateSnippets;

    return (
      <DashbordWrapper>
        { isEmpty(this.state.searchTags) && (
          <React.Fragment>
            <Private style={ this.linearGradient(privateSnippets) }
                     onClick={ () => searchByStatus('private') }>
              <h3>Private</h3>
              <span>
                { privateSnippets }
              </span>
            </Private>

            <Public style={ this.linearGradient(publicSnippetsCount) }
                     onClick={ () => searchByStatus('public') }>
              <h3>Public</h3>
              <span>
                { publicSnippetsCount }
              </span>
            </Public>

            <Starred style={ this.linearGradient(starred) }
                     onClick={ () => searchByStatus('starred') }>
              <h3>Starred</h3>
              <span>
                { starred }
              </span>
            </Starred>

            <Untitled style={ this.linearGradient(this.getUntitled()) }
                      onClick={ () => searchByStatus('untitled') }>
              <h3>Untitled:</h3>
              <span>
                { this.getUntitled() }
              </span>
            </Untitled>
          </React.Fragment>
        ) }

        { isEmpty(this.state.searchTags) && (
          <Language>
            <h3>Languages:</h3>
            <ScrollPad>
              <div>
                { this.renderLanguages() }
              </div>
            </ScrollPad>
          </Language>
        ) }

        { isEmpty(this.state.searchTags) && (
          <Stars>
            <HeadingWithSearch>
              <h3>Starred:</h3>
              <div>
                <Icon type="search" size="22" color={ baseAppColor }/>
                <StyledInput type="search"
                             placeholder="Search starred"
                             onChange={ (event) => this.searchStarred(event.target.value) }/>
              </div>
            </HeadingWithSearch>
            <ScrollPad>
              <ul>
                { this.renderStarred() }
              </ul>
            </ScrollPad>
          </Stars>
        ) }

        <Tags>
          <HeadingWithSearch>
            <h3>Tags:</h3>
            <div>
              <Icon type="search" size="22" color={ baseAppColor }/>
              <StyledInput type="search"
                           placeholder="Search tags"
                           onChange={ (event) => this.searchTags(event.target.value) }/>
            </div>
          </HeadingWithSearch>
          <ScrollPad>
            <div>
              { this.renderTags() }
            </div>
          </ScrollPad>
        </Tags>
      </DashbordWrapper>
    );
  }
}

const mapStateToProps = (state) => ({
  snippets: getSnippets(state),
  starred: getStarredCount(state),
  snippetsLanguages: getLanguages(state),
  privateSnippets: getPrivate(state)
});

DashBoard.propTypes = {
  snippets: PropTypes.object,
  starred: PropTypes.number,
  searchByTags: PropTypes.func,
  searchByLanguages: PropTypes.func,
  searchByStatus: PropTypes.func,
  getRateLimit: PropTypes.func,
  snippetsLanguages: PropTypes.array,
  privateSnippets: PropTypes.number
};

export default connect(mapStateToProps, {
  searchByTags: snippetActions.filterSnippetsByTags,
  searchByLanguages: snippetActions.filterSnippetsByLanguage,
  searchByStatus: snippetActions.filterSnippetsByStatus,
  getRateLimit: snippetActions.getRateLimit
})(DashBoard);
