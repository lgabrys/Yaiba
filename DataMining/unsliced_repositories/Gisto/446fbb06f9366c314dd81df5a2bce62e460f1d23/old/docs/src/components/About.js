import React from 'react';
import { Link } from 'gatsby';
import Logo from './Logo';

const About = () => (
  <div className="w-container">
    <div className="w-row">

      <div className="w-col w-col-4 w-clearfix">
        <h3><b>why</b></h3>

        <p>
          <Logo/> started by fulfilling a lack of a syntax highlighted and cloud
          synchronized code
          snippet solution.
        </p>
      </div>

      <div className="w-col w-col-4 w-clearfix the-who">
        <h3><b>who</b></h3>

        <p>
          <Logo/> is developed by a team of two web developers on their spare
          time.
        </p>

        <p>
          We appreciate any suggestions or contributions to make <Logo/> better.
        </p>

      </div>

      <div className="w-col w-col-4 w-clearfix">
        <h3><b>how</b></h3>

        <p>
          <Logo/> is built using the open web technologies using several open
          source projects such as
          React, Electron, Monaco Editor and many more
        </p>
      </div>

    </div>
  </div>
);

export default About;
