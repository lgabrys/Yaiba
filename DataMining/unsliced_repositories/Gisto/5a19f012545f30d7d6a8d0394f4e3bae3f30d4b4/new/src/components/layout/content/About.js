import React from 'react';
import styled from 'styled-components';

import ExternalLink from 'components/common/ExternalLink';

import * as packageJson from '../../../../package.json';
import logoImg from '../../../../build/icon.png';

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 50%;
  margin: 0 auto;
`;

export const About = () => (
  <Wrapper>
    <img src={ logoImg } width="80" alt=""/>
    <h2>About Gisto</h2>
    <p>Current version <strong>v{packageJson.version}</strong></p>

    <p>
      Gisto is a code snippet manager that runs on GitHub Gists and adds additional features such as
      searching,
      tagging and sharing gists while including a rich code editor.
    </p>

    <p>
      All your data is stored on GitHub and you can access it from GitHub Gists at any time with
      changes carrying over to Gisto.
    </p>

    <p>
      <ExternalLink href="https://github.com/Gisto/Gisto">GitHub</ExternalLink>&nbsp;|&nbsp;
      <ExternalLink href="https://gistoapp.com">Website gistoapp.com</ExternalLink>&nbsp;|&nbsp;
      <ExternalLink href="https://github.com/Gisto/Gisto/issues">Issues</ExternalLink>&nbsp;|&nbsp;
      <ExternalLink href="https://twitter.com/gistoapp">Twitter</ExternalLink>
    </p>

  </Wrapper>
);

About.propTypes = {};

export default About;
