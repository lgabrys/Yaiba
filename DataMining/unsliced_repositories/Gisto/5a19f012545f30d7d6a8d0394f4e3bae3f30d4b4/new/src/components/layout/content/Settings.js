import React from 'react';

export const Settings = () => (
  <div>Settings</div>
);

Settings.propTypes = {};

export default Settings;
