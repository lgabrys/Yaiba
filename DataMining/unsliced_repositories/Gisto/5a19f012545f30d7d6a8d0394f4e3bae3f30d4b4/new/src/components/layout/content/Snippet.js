import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import styled from 'styled-components';
import { get, map, filter } from 'lodash/fp';

import * as snippetActions from 'actions/snippets';
import { borderColor } from 'constants/colors';

import Editor from 'components/common/Editor';
import SnippetHeader from 'components/layout/content/snippet/snippetHeader';

import 'github-markdown-css/github-markdown.css';

const SnippetWrapper = styled.div`
  background: #fff;
  font-weight: 200;
  width: auto;
  display: flex;
  flex-direction: column;
  border-radius: 3px;
  box-shadow: 0 0 10px ${borderColor};
  flex: 1;
  margin-bottom: 20px;
`;

export class Snippet extends React.Component {
  componentDidMount() {
    this.props.getSnippet(this.props.match.params.id || this.props.snippet.id);
  }

  componentDidUpdate(prevProps) {
    if (prevProps.match.params.id !== this.props.match.params.id) {
      this.props.getSnippet(this.props.match.params.id || this.props.snippet.id);
    }
  }

  render() {
    const {
      snippet, edit, tempSnippet, updateTempSnippet
    } = this.props;
    const currentSnippet = edit ? tempSnippet : snippet;
    const files = filter((file) => !file.delete, get('files', currentSnippet));

    if (!this.props.snippet) {
      return null;
    }

    return (
      <React.Fragment>
        {  map((file) => (
          <SnippetWrapper key={ file.uuid || `${file.size}-${file.viewed}-${file.filename}` }>
            <SnippetHeader file={ file }
                           username={ snippet.username }
                           snippetId={ snippet.id }/>
            <Editor file={ file }
                    edit={ edit }
                    onChange={ (value) => updateTempSnippet(['files', file.uuid, 'content'], value) }/>
          </SnippetWrapper>
        ), files) }
      </React.Fragment>
    );
  }
}

const mapStateToProps = (state, ownProps) => {
  const snippetId = ownProps.match.params.id;

  return {
    snippet: get(['snippets', 'snippets', snippetId], state),
    tempSnippet: get(['snippets', 'edit'], state),
    edit: get(['ui', 'snippets', 'edit'], state)
  };
};

Snippet.propTypes = {
  match: PropTypes.object,
  snippet: PropTypes.object,
  getSnippet: PropTypes.func.isRequired,
  edit: PropTypes.bool,
  tempSnippet: PropTypes.object,
  updateTempSnippet: PropTypes.func
};

export default connect(mapStateToProps, {
  getSnippet: snippetActions.getSnippet,
  updateTempSnippet: snippetActions.updateTempSnippet
})(Snippet);
