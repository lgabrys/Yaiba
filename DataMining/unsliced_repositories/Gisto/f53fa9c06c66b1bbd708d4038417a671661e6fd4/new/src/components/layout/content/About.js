import React from 'react';

const About = () => (
  <div>About</div>
);

About.propTypes = {};

export default About;
