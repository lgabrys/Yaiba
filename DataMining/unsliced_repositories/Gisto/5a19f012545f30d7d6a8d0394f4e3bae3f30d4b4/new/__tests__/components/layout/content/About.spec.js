import { About } from 'components/layout/content/About';

jest.mock('../../../../build/icon.png', () => 'icon.png');

const propSetup = (props) => ({
  ...props
});

const setup = (props) => mount(<About { ...propSetup(props) }/>);

describe('COMPONENTS - <About>', () => {
  test('render About', () => {
    const component = setup();

    expect(component).toMatchSnapshot();
  });
});
