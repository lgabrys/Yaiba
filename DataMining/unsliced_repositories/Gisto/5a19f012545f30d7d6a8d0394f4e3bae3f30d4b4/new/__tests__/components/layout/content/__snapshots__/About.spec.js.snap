// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENTS - <About> render About 1`] = `
<About>
  <Wrapper>
    <div
      className="Wrapper-dyuznp kuhFqe"
    >
      <img
        alt=""
        src="icon.png"
        width="80"
      />
      <h2>
        About Gisto
      </h2>
      <p>
        Current version 
        <strong>
          v
          1.9.36
        </strong>
      </p>
      <p>
        Gisto is a code snippet manager that runs on GitHub Gists and adds additional features such as searching, tagging and sharing gists while including a rich code editor.
      </p>
      <p>
        All your data is stored on GitHub and you can access it from GitHub Gists at any time with changes carrying over to Gisto.
      </p>
      <p>
        <ExternalLink
          href="https://github.com/Gisto/Gisto"
        >
          <Anchor
            href="#"
            onClick={[Function]}
          >
            <Link
              href="#"
              onClick={[Function]}
            >
              <a
                className="Link-hdnizH bVHTue"
                href="#"
                onClick={[Function]}
              >
                GitHub
              </a>
            </Link>
          </Anchor>
        </ExternalLink>
         | 
        <ExternalLink
          href="https://gistoapp.com"
        >
          <Anchor
            href="#"
            onClick={[Function]}
          >
            <Link
              href="#"
              onClick={[Function]}
            >
              <a
                className="Link-hdnizH bVHTue"
                href="#"
                onClick={[Function]}
              >
                Website gistoapp.com
              </a>
            </Link>
          </Anchor>
        </ExternalLink>
         | 
        <ExternalLink
          href="https://github.com/Gisto/Gisto/issues"
        >
          <Anchor
            href="#"
            onClick={[Function]}
          >
            <Link
              href="#"
              onClick={[Function]}
            >
              <a
                className="Link-hdnizH bVHTue"
                href="#"
                onClick={[Function]}
              >
                Issues
              </a>
            </Link>
          </Anchor>
        </ExternalLink>
         | 
        <ExternalLink
          href="https://twitter.com/gistoapp"
        >
          <Anchor
            href="#"
            onClick={[Function]}
          >
            <Link
              href="#"
              onClick={[Function]}
            >
              <a
                className="Link-hdnizH bVHTue"
                href="#"
                onClick={[Function]}
              >
                Twitter
              </a>
            </Link>
          </Anchor>
        </ExternalLink>
      </p>
    </div>
  </Wrapper>
</About>
`;
